src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -782,18 +716,19 @@ int D4NFilterObject::D4NFilterReadOp::D4NFilterGetCB::handle_data(bufferlist& bl
       std::string oid = this->oid + "_" + std::to_string(ofs) + "_" + std::to_string(bl_len);
       ret = filter->get_cache_driver()->put_async(save_dpp, oid, bl, bl.length(), source->get_attrs());
       if (ret == 0) {
-        block.size = bl.length();
-        block.blockId = oid;
-        block.hostsList.push_back(blockDir->get_addr().host + ":" + std::to_string(blockDir->get_addr().port)); //check if this is the correct way to get the local addr
-        block.size = source->get_obj_size();
-        block.cacheObj.bucketName = source->get_bucket()->get_name();
+        block.version = oid;
+        block.size = bl.length(); // TODO: fill out block entirely
+        block.size = source->get_obj_size(); // which is correct? -Sam
```

Comment:
block.size is not obj_size() atleast in this case. While saving the blocks to the cache for a 'read' cache, the object is being divided up in blocks, and the blocks are being saved in 4 MB chunks which != obj_size(). Also the last chunk isn't always 4 MB in size, it can be lesser than that.